Pharma Sales Force Effectiveness Analytics System

Objective:
Analyze sales performance, territory effectiveness,
physician contribution, product performance,
and marketing ROI to support data-driven decisions.

Business Questions:
1. Which regions generate highest revenue?
2. Which sales reps exceed targets?
3. Which doctors generate maximum prescriptions?
4. Which medicines contribute most revenue?
5. Which regions generate highest ROI?

In [68]:
import pandas as pd
import numpy as np

In [69]:
df = pd.read_csv("/Users/saurabhpatankar/Desktop/pharma-sales-analytics/data/pharma_sales_data.csv")

df.head()

,sales_rep_id,sales_rep_name,doctor_id,doctor_name,specialty,region,medicine_name,units_sold,revenue,visit_date,target,achieved_pct,marketing_spend,prescriptions,roi
0,1001,Jessica Vaughn,2010,Allison Jones,Neurology,South,NeuroPlus,62,78616,2025-08-03,183348,42.88,11972,47,6.57
1,1003,Joseph Schroeder,2009,Valerie White,Cardiology,South,HeartSafe,48,31152,2026-02-28,93509,33.31,5160,9,6.04
2,1011,Krystal Reese,2056,Michael Smith,Dermatology,South,SkinGlow,306,80478,2025-12-03,193999,41.48,32192,17,2.50
3,1015,Justin Hays,2016,Isaac Mann,Cardiology,South,HeartSafe,242,70664,2025-09-03,55555,127.20,27263,44,2.59
4,1004,Jordan Adams,2053,Jason Meyers,Neurology,South,NeuroPlus,161,186277,2025-06-24,181653,102.55,37583,100,4.96


DATA UNDERSTANDING


In [70]:
df.shape

(2000, 15)

In [71]:
df.columns

Index(['sales_rep_id', 'sales_rep_name', 'doctor_id', 'doctor_name',
       'specialty', 'region', 'medicine_name', 'units_sold', 'revenue',
       'visit_date', 'target', 'achieved_pct', 'marketing_spend',
       'prescriptions', 'roi'],
      dtype='str')

In [72]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   sales_rep_id     2000 non-null   int64  
 1   sales_rep_name   2000 non-null   str    
 2   doctor_id        2000 non-null   int64  
 3   doctor_name      2000 non-null   str    
 4   specialty        2000 non-null   str    
 5   region           2000 non-null   str    
 6   medicine_name    2000 non-null   str    
 7   units_sold       2000 non-null   int64  
 8   revenue          2000 non-null   int64  
 9   visit_date       2000 non-null   str    
 10  target           2000 non-null   int64  
 11  achieved_pct     2000 non-null   float64
 12  marketing_spend  2000 non-null   int64  
 13  prescriptions    2000 non-null   int64  
 14  roi              2000 non-null   float64
dtypes: float64(2), int64(7), str(6)
memory usage: 234.5 KB


In [73]:
df.describe()

,sales_rep_id,doctor_id,units_sold,revenue,target,achieved_pct,marketing_spend,prescriptions,roi
count,2000.00000,2000.000000,2000.000000,2000.000000,2000.000000,2000.000000,2000.000000,2000.000000,2000.000000
mean,1007.59950,2030.068000,257.512500,218771.312500,125499.740000,200.173565,22465.853000,52.631000,12.523540
std,4.52751,17.287765,138.433405,160440.190352,43156.652422,178.555146,10059.869943,27.302886,13.164621
min,1000.00000,2000.000000,20.000000,4774.000000,50077.000000,3.230000,5055.000000,5.000000,0.220000
25%,1004.00000,2015.750000,137.000000,90541.250000,88944.750000,73.375000,13740.250000,30.000000,4.210000
50%,1008.00000,2030.000000,257.500000,181620.000000,124910.000000,149.855000,22353.500000,53.000000,8.420000
75%,1011.00000,2046.000000,377.250000,314291.250000,162841.500000,273.095000,31185.500000,75.000000,15.902500
max,1015.00000,2059.000000,500.000000,745500.000000,199961.000000,1337.350000,39996.000000,100.000000,142.140000


DATA QUALITY CHECKS


In [74]:
df.isnull().sum()

sales_rep_id       0
sales_rep_name     0
doctor_id          0
doctor_name        0
specialty          0
region             0
medicine_name      0
units_sold         0
revenue            0
visit_date         0
target             0
achieved_pct       0
marketing_spend    0
prescriptions      0
roi                0
dtype: int64

In [75]:
df.duplicated().sum()

np.int64(0)

In [76]:
(df["revenue"] < 0).sum()

np.int64(0)

In [77]:
(df["marketing_spend"] < 0).sum()

np.int64(0)

KPI CALCULATION


In [78]:
total_revenue = df["revenue"].sum()

In [79]:
total_prescriptions = df["prescriptions"].sum()

In [80]:
avg_achievement = df["achieved_pct"].mean()

In [81]:
avg_roi = df["roi"].mean()

In [82]:
kpi_df = pd.DataFrame({
    "Metric":[
        "Revenue",
        "Prescriptions",
        "Achievement %",
        "ROI"
    ],
    "Value":[
        total_revenue,
        total_prescriptions,
        avg_achievement,
        avg_roi
    ]
})

kpi_df

,Metric,Value
0,Revenue,4.375426e+08
1,Prescriptions,1.052620e+05
2,Achievement %,2.001736e+02
3,ROI,1.252354e+01


Revenue by Region

In [83]:
revenue_region = (
    df.groupby("region")["revenue"]
      .sum()
      .sort_values(ascending=False)
)

revenue_region

region
South    115117327
East     110602001
West     108383907
North    103439390
Name: revenue, dtype: int64

Monthly Revenue Trend

In [84]:
df["visit_date"] = pd.to_datetime(df["visit_date"])

monthly_revenue = (
    df.groupby(df["visit_date"].dt.to_period("M"))
      ["revenue"]
      .sum()
)

monthly_revenue

visit_date
2025-05     9375621
2025-06    41602915
2025-07    35748771
2025-08    39507879
2025-09    29435432
2025-10    42338071
2025-11    34488537
2025-12    39990121
2026-01    43385908
2026-02    31885635
2026-03    35865896
2026-04    28638054
2026-05    25279785
Freq: M, Name: revenue, dtype: int64

Territory Contribution %

In [85]:
territory = (
    df.groupby("region")["revenue"]
      .sum()
      .reset_index()
)

territory["Contribution_%"] = (
    territory["revenue"]
    /
    territory["revenue"].sum()
    *100
)

territory

,region,revenue,Contribution_%
0,East,110602001,25.277995
1,North,103439390,23.640986
2,South,115117327,26.309969
3,West,108383907,24.771051


Top Reps

In [86]:
top_reps = (
    df.groupby("sales_rep_name")
      .agg({
          "revenue":"sum",
          "achieved_pct":"mean"
      })
      .sort_values(
          "revenue",
          ascending=False
      )
)

top_reps

,revenue,achieved_pct
sales_rep_name,,
Michelle Francis,36520998,221.956013
Daniel Thompson,33252051,208.907483
Robert Pace,31434496,218.068088
William Levy,30523089,213.300977
Ruth Holmes,30441959,203.086715
Justin Hays,29766503,187.696312
Megan Terry,29175846,210.266466
Joanna Smith,29032346,212.156439
Krystal Reese,28285060,181.997319


Underperforming Reps

In [87]:
top_reps[
    top_reps["achieved_pct"] < 180
]

,revenue,achieved_pct
sales_rep_name,,
Jordan Adams,24908585,175.144724


Top Doctors

In [88]:
doctor_analysis = (
    df.groupby("doctor_name")
      ["prescriptions"]
      .sum()
      .sort_values(
          ascending=False
      )
)

doctor_analysis

doctor_name
Roy Newman                2818
Michael Daniels           2621
Nicole Hernandez          2550
Andrew Moore              2493
Justin Hill               2367
Tiffany Black             2234
Becky Garza               2164
Eric Perez                2087
William Rios              2060
Michael Robertson         2003
Katherine Gregory         1986
Christopher Davis         1967
Seth Evans                1957
Dean Johnson              1955
Tracy Williams            1936
Kathleen Harris           1898
Tina Lara                 1894
Nathan Buckley            1881
Jacob Wright              1861
Valerie Gaines            1860
Andrew Matthews           1858
Isaac Mann                1803
Michael Smith             1797
Allison Quinn             1792
Paula Martinez            1789
Cynthia Hodges            1767
Jacob Juarez              1764
Brandon Rodriguez         1748
Gina Hernandez            1731
Gerald Richardson         1725
Sara Salazar              1719
Elijah Thompson II        1

Doctor Segmentation

In [89]:
doctor_df = (
    df.groupby("doctor_name")
      ["prescriptions"]
      .sum()
      .reset_index()
)

doctor_df["Segment"] = np.where(
    doctor_df["prescriptions"] > 2500,
    "High Value",
    np.where(
        doctor_df["prescriptions"] > 1500,
        "Medium Value",
        "Low Value"
    )
)

doctor_df

,doctor_name,prescriptions,Segment
0,Adam Bonilla,1619,Medium Value
1,Allison Jones,1393,Low Value
2,Allison Quinn,1792,Medium Value
3,Allison Zimmerman,1560,Medium Value
4,Andrew Huffman,1470,Low Value
5,Andrew Matthews,1858,Medium Value
6,Andrew Moore,2493,Medium Value
7,Angela Thomas,1567,Medium Value
8,Anna Thomas,1575,Medium Value
9,Barry Daniel,1252,Low Value


Product Revenue

In [90]:
product_revenue = (
    df.groupby("medicine_name")
      ["revenue"]
      .sum()
      .sort_values(
          ascending=False
      )
)

product_revenue

medicine_name
HeartSafe    140199890
NeuroPlus    100603615
SkinGlow      99043425
PainOff       97695695
Name: revenue, dtype: int64

ROI by Region

In [91]:
roi_region = (
    df.groupby("region")
      .agg({
          "revenue":"sum",
          "marketing_spend":"sum"
      })
)

roi_region["roi"] = (
    roi_region["revenue"]
    /
    roi_region["marketing_spend"]
)

roi_region.sort_values(
    "roi",
    ascending=False
)

,revenue,marketing_spend,roi
region,,,
North,103439390,10125908,10.215320
East,110602001,11401471,9.700678
South,115117327,11954748,9.629423
West,108383907,11449579,9.466191


# Executive Summary & Business Recommendations

The Pharma Sales Force Effectiveness Analytics project was conducted to evaluate territory performance, sales representative effectiveness, physician contribution, product performance, and marketing ROI.

Analysis revealed that the South region generated the highest revenue (₹115.1M), indicating strong market penetration and commercial performance. However, the North region achieved the highest marketing ROI (10.22), demonstrating the most efficient utilization of marketing investments. This highlights the importance of balancing revenue growth with operational efficiency.

Physician analysis identified Roy Newman as the highest-value physician based on prescription volume, suggesting that a small group of physicians contributes significantly to overall business performance. Product analysis showed that HeartSafe is the leading revenue-generating medicine, making it a critical driver of portfolio growth.

Sales force analysis indicated strong overall performance, with all regions exceeding their assigned targets and East region achieving the highest average target attainment. This suggests effective field execution but also indicates that target-setting methodologies may require reassessment to ensure continued performance improvement.

### Key Recommendations

* Continue investing in high-revenue territories while replicating their successful strategies across other regions.
* Increase engagement with high-value physicians to maximize prescription generation and strengthen relationships.
* Optimize marketing investments by adopting best practices from high-ROI regions.
* Prioritize promotion and availability of top-performing products such as HeartSafe.
* Reassess sales targets and introduce stretch goals to drive continuous improvement in sales force performance.

This project demonstrates how pharmaceutical sales data can be transformed into actionable business insights that support territory optimization, physician targeting, sales force effectiveness, and strategic decision-making.
